notes_app/client/app/notes/notes_list.component.js

```diff
@@ -15,11 +16,11 @@
         notesService.pagenumToOffset(ctrl.currentPage, ctrl.noteLimit)
       ).then(function(response) {
         ctrl.metadata = response.meta;
-        ctrl.notes = response.objects;
+        ctrl.notes = notesService.data.notes;
```

Comment:
it can't work out of ```then()``` because we set ```notesService.data``` object asynchronously. so that is why here we set ```ctrl.notes``` and ```ctrl.metadata```  in ```then()``` also